wasi.go

```diff
@@ -57,7 +57,7 @@ func WASISnapshotPreview1WithConfig(c *WASIConfig) *HostFunctions {
 		opts = append(opts, opt)
 	}
 	if len(c.Environ) > 0 {
-		environ := make([]string, len(c.Environ))
+		environ := make([]string, 0)
```

Comment:
Thanks and sorry about the bug!
```suggestion
		environ := make([]string, 0, len(c.Environ))
```